docs/proposals/config_data.md

```diff
@@ -0,0 +1,316 @@
+
+1. `ConfigData` object is created through the API server.
+2. New containers using the `ConfigData` object are created.
+3. `ConfigData` object is retrieved using the downward API.
+4. `Data` key/value pairs of `ConfigData` will be exposed as environment variables (`EnvVarSource`).
+
+### Mounted Volume
+
+The container configuration workflow is given below:
+
+1. `ConfigData` object is created through the API server.
+2. New containers using the `ConfigData` object are created.
+3. `ConfigData` object is retrieved using the downward API.
+4. `ConfigData` will be mounted as a new volume (`ConfigDataVolumeSource`) where keys are file names and values are file contents.
+5. API server is watched for modifications to the object.
+6. After a modification, actions 3, 4, and 5 are repeated.
+
+## Examples
+
+#### Consuming ConfigData as volumes
+
+`redis-volume-config` is intended to be used as a mounted volume containing two files with their respective contents.
+```json
+{
+    "apiVersion": "extensions/v1beta1",
+    "metadata": {
+        "name": "redis-volume-config",
+    },
+    "kind": "ConfigData",
+    "data": {
+        "redis.conf": "pidfile /var/run/redis.pid\nport6379\ntcp-backlog 511\n databases 1\ntimeout 0\n",
+        "slave_run.sh": "#!/bin/bash\nredis-server --slaveof ${REDISMASTER_SERVICE_HOST:-$SERVICE_HOST} $REDISMASTER_SERVICE_PORT\n",
+        "redis_dockerbuild.sh": "docker build -t redis\ndocker build -t redis-master\ndocker build -t redis-slave\n"
+    }
+}
+```
+
+Configuration objects are persisted to the API server as follows:
+```bash
+kubectl create -f pod-configuration-vol.json 
+```
+
+ExampleReplicationController starts 2 replicas that runs the `busybox` container.
+In this use case, the `ConfigData` object will be mounted as a volume and contain
+four files named respectively "redis.conf", "slave_run.sh", and "redis_dockerbuild.sh".
+
+```yaml
+apiVersion: extensions/v1beta1
+kind: ReplicationController
+metadata:
+  name: ExampleReplicationController
+spec:
+  replicas: 2
+  selector:
+    app: busybox
+  template:
+    metadata:
+      labels:
+        app: busybox
+    spec:
+      containers:
+        - name: busybox
+          image: gcr.io/google_containers/busybox
+          ports:
+            - containerPort: 80
+          volumeMounts:
+            - name: config-data-volume
+              mountPath: /mnt/config-data
+      volumes:
+      - name: config-data-volume
+        configDataVolumes:
+          items:
+            - path: "etc/redis.conf"
+              configDataRef:
+                from: redis-volume-config
+                fieldPath: redis.conf
+            - path: "scripts/slave_run.sh"
+              configDataRef:
+                from: redis-volume-config
+                fieldPath: slave_run.sh
+            - path: "scripts/redis_dockerbuild.sh"
+              configDataRef:
+                from: redis-volume-config
+                fieldPath: redis_dockerbuild.sh
+```
```

Comment:
@gurvindersingh Templating will not be part of this proposal.  At this moment, what you see here is basically what will be in this proposal.  We're working now on getting the details just right so this proposal can go in.